src/main/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataService.java

```diff
@@ -49,8 +49,9 @@ public void upsertMetadataFor(Event event) {
                         Iterator<String> metadataKeys = eventDataNode.get("external_metadata").fieldNames();
                         metadataKeys.forEachRemaining(metadataKey -> {
                             metadataKeyDao.insertIfNotExist(metadataKey);
+                            String value = eventDataNode.get("external_metadata").get(metadataKey).textValue();
```

Comment:
We need to get the metadata from the EventStream so that we can re-project old transactions to back-fill.
This will work for new transactions though, will using the EventStream be the next step?

Reply:
It's small iterations, next will be reworking the method to accept an EventDigest object and work from that.

Reply:
👍